functions/go/create-setters/createsetters/create_setters.go

```diff
@@ -0,0 +1,357 @@
+	Results []*Result
+
+	// filePath file path of resource
+	filePath string
+}
+
+// ScalarSetter stores name and value of the map setter
+type ScalarSetter struct {
+	// Name is the name of the setter
+	Name string
+
+	// Value is the value of the field to which setter comment is added.
+	Value string
+}
+
+// ArraySetter stores name and values of the array setter
+type ArraySetter struct {
+	// Name is the name of the setter
+	Name string
+
+	// Values are the values of the field to which setter comment is added.
+	Values []string
+}
+
+// Result holds result of create-setters operation
+type Result struct {
+	// FilePath is the file path of the matching value
+	FilePath string
+
+	// FieldPath is field path of the matching value
+	FieldPath string
+
+	// Value is the value of the field to which setter comment is added.
+	Value string
+
+	// Comment is the line comment of the matching value
+	Comment string
+}
+
+// CompareSetters is to sort the setter values
+type CompareSetters []ScalarSetter
+
+func (a CompareSetters) Len() int {
+	return len(a)
+}
+
+// checks if the jth node's Name is a substring of ith node's Value
+// jth node is sorted first if the condition statisfies
+func (a CompareSetters) Less(i, j int) bool {
+	return !strings.Contains(a[i].Value, a[j].Name)
+}
+
+func (a CompareSetters) Swap(i, j int) {
+	a[i], a[j] = a[j], a[i]
+}
+
+// Filter implements CreatSetters cs a yaml.Filter
+func (cs *CreateSetters) Filter(nodes []*yaml.RNode) ([]*yaml.RNode, error) {
+
+	for i := range nodes {
+		filePath, _, err := kioutil.GetFileAnnotations(nodes[i])
+		if err != nil {
+			return nodes, err
+		}
+		cs.filePath = filePath
+		err = accept(cs, nodes[i])
+		if err != nil {
+			return nil, errors.Wrap(err)
+		}
+	}
+	return nodes, nil
+}
+
+/**
+visitMapping takes the mapping node and performs following steps,
+checks if it is a sequence node
+checks if all the values in the node match any of the ArraySetters
+adds the linecomment if they are equal
+
+checks if any of the values of node matches with ScalarSetters
+changes the node to FoldedStyle
+
+e.g. for input of Mapping node
+
+environments:
+  - dev
+  - stage
+
+For input CreateSetters [Name: env, Values: [dev, stage]], yaml node is transformed to
+
+environments: # kpt-set: ${env}
+  - dev
+  - stage
+
+e.g. for input of Mapping node with FlowStyle
+
+env: [foo, bar]
+
+For input CreateSetters [Name: env, Values: [foo, bar]], yaml node is transformed to
+
+env: [foo, bar] # kpt-set: ${env}
+
+e.g. for input of Mapping node with FlowStyle matching few values from ScalarSetters
+
+env: [foo, bar]
+
+For input CreateSetters [Name: image, Values: foo], yaml node is transformed to
+
+env:
+  - foo
+  - bar
+*/
+
+func (cs *CreateSetters) visitMapping(object *yaml.RNode, path string) error {
+	return object.VisitFields(func(node *yaml.MapNode) error {
+		if node == nil || node.Key.IsNil() || node.Value.IsNil() {
+			// don't do IsNilOrEmpty check cs empty sequences are allowed
```

Comment:
What does `cs` mean here?

Reply:
typo in the comment, changed it to `as`.